Added a check if LD is used without methods that allow it

rapoliveira opened the pull request:
The main contribution here is a new function called `_check_limb_darkening()`.
In models with limb darkening, it does not allow the use of only uniform methods (finite_source_uniform_X, with X being Gould94, Gould94_direct, WittMao94, or Lee09), requiring at least one method that includes LD.

Two new tests checks if the function works well for single source and binary source. For the binary source, a different limb darkening value is tested for each source, including a zero value.

An additional test covers the five errors raised in `set_magnification_methods()`, which were not covered by any tests.

rapoliveira commented:
@rpoleski My last commits add tests for multiple sources and cover the ValueError of the new function added by you.

rpoleski commented:
Looks good. Please use `coverage` tool to make sure that all code add to `model.py` is covered by tests. Also please update version to 3.1.3.

rapoliveira commented:
@rpoleski the added lines were covered by tests, but now I added an extra test for the lines missing in `get_magnification()` and `_get_magnification()` functions. Version number is also updated.

Side question: is there a way to test the current line 1334 of model.py, with `self.n_sources < 1`?

rpoleski commented:
Side answer (i.e., answer to a side question): not, unless a user starts changing private properties. Though, if one changes `ModelParameters._count_sources()`, then the error may be produced with `.n_sources < 0`. An approach in which such cases are considered is called "defensive programming."

Merging.
